keras_nlp/models/gemma/gemma_causal_lm.py

```diff
@@ -283,19 +284,19 @@ def next(prompt, cache, index):
             cache=cache,
             index=index,
             mask=padding_mask,
-            end_token_id=end_token_id,
+            stop_token_ids=stop_token_ids,
             hidden_states=hidden_states,
             model=self,
         )
 
         # Compute an output padding mask with the token ids we updated.
-        if end_token_id is not None:
-            # Build a mask of `end_token_id` locations not in the original
+        if stop_token_ids is not None:
+            # Build a mask of `stop_token_ids` locations not in the original
             # prompt (not in locations where `padding_mask` is True).
-            end_locations = ops.logical_and(
-                ops.equal(token_ids, end_token_id),
-                ops.logical_not(padding_mask),
+            end_locations = any_equal(
```

Comment:
LGTM either way, but slight preference for the clarify of keeping the padding mask split out...

```python
end_locations = any_equal(token_ids, stop_token_ids)
# Only consider newly created stop tokens.
new_locations = ops.logical_not(padding_mask)
end_locations = ops.logical_and(stop_locations, new_locations)
```

Other wise we invert the padding mask, but still refer to it as a padding mask in the utility docs. Just seems harder to follow.

Reply:
ACK - leaving for now if no strong preference